tests/Bundle/ServiceInstantiationTest.php

```diff
@@ -121,9 +122,10 @@ public function testItCanMapEnums(): void
     /**
      * This test validates that PropertyInfoPass is correctly applied.
      */
-    public function testMapClassWithPrivateProperty(): void
+    public function testMapToClassWithPrivateProperty(): void
     {
-        static::bootKernel();
+        static::bootKernel(['additionalConfigFile' => __DIR__ . '/Resources/config/with-private-properties.yml']);
```

Comment:
when making `ReflectionExtractor` aware of `map_private_properties`, it has to be set to `true` for this test to pass.

but it seems wrong to me: even if the property is private, it could be written, because it is in the constructor:
```php
class ClassWithPrivateProperty
{
    public function __construct(
        private string $foo
    ) {
    }
}
```

Reply:
:warning: We need to fix this behavior before merging this pull request !

Reply:
It depends, it only work if we write to it, but not if we read from it

Reply:
> it only work if we write to it, but not if we read from it

it is indeed IMO the expected behavior, even with `map_private_properties: false`, but that's not what happens

Reply:
Yes, but not sure we can resolve this easily on our side ?

Does this bug happens because we cannot found the property (not listed) or because we don't find a mutator for it ?

AFAIK, this would require having a different service when we try to get the write mutator on constructor, so we could configure that one to work with private properties

Or do a bug fix in symfony ?

Reply:
ok, forget about that all :)

I've found [a simpler solution](https://github.com/jolicode/automapper/pull/137)